docs/developer-guide/api-overview.md

```diff
@@ -20,153 +20,154 @@ The relevant modules are available in Maven Central:
                 .body("pong", UTF_8)
                 .build();
         
         @Override
-        protected HttpResponse doHandle(HttpRequest request) {
-            return PONG.toStreamingResponse();
+        protected LiveHttpResponse doHandle(LiveHttpRequest request) {
+            return PONG.stream();
         }
     }   
 ```
  
 ### Http Interceptor Interface
 
-A HTTP interceptor is an object that transforms, responds, or runs side-effecting actions 
-for HTTP traffic passing through. 
-
-Styx arranges interceptors in a linear chain, forming a core of its proxying pipeline.
-All recieved traffic goes through the interceptor chain which then acts on the traffic
-accordingly.  
+Styx has a HTTP pipeline that processes all received HTTP messages.
+The pipeline is made of a linear chain of HTTP interceptors. An HTTP interceptor 
+transforms, responds, or runs side-effecting actions on a live HTTP traffic.
   
-Styx has some internal Styx interceptors. But normally it is the custom plugins that
-add value for Styx deployments. An extension point for a custom plugin is the 
-`HttpInterceptor` interface. It has only one method:
+Styx provides some internal interceptors. You can also implement own
+interceptor plugins. That's what Styx is for! An extension point for a 
+custom plugin is the `HttpInterceptor` interface. It has only one method:
 
 ```java
     public interface HttpInterceptor {
        ...
-       StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain);
+       Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain);
+       ...
     }
 ```
 
-A Styx plugin is an implementation of this method. 
-
-It is the `intercept` method which transforms or acts on a received request, 
-and its corresponding response. As an event based system, all implementations 
-must be strictly non-blocking. Blocking the thread would stall Styx event processing 
-loop. So take care to stick with asynchronous implementation.
+The interface shows that `intercept` method receives a live HTTP request, 
+and eventually returns a live HTTP response. 
 
-The received request is passed in as its first argument. The second argument, `Chain`, is
-a handle to the remaining tail of the `HttpInterceptor` chain. The most important
-function of the chain is the `proceed` method. It passes the request to the
-next interceptor in the chain. It returns a response observable, in which you
-bind any response transformations. For example:
+The `intercept` may transform or run side effecting actions on request or response objects. 
+As an event based system, all implementations must be strictly non-blocking. 
+Blocking a thread stalls the Styx event processing loop. So take care to 
+stick with asynchronous implementation.
 
+A HTTP request is passed in as its first argument. The second argument, `Chain`, is
+a handle to the next `HttpInterceptor` in the pipeline. The `Chain` has a `proceed` mehtod
```

Comment:
`mehtod` is misspelt. Should be `method`.

Reply:
👍